graphistry/pygraphistry.py

```diff
@@ -242,12 +245,63 @@ def sso_login(org_name=None, idp_name=None, sso_timeout=SSO_GET_TOKEN_ELAPSE_SEC
         except Exception:  # required to log on
             # print("required to log on")
             PyGraphistry.sso_state(arrow_uploader.sso_state)
-
             auth_url = arrow_uploader.sso_auth_url
             # print("auth_url : {}".format(auth_url))
             if auth_url and not PyGraphistry.api_token():
                 PyGraphistry._handle_auth_url(auth_url, sso_timeout, sso_opt_into_type)
-                return auth_url
+  
+    @staticmethod
+    def databricks_notebook_sso_login( 
+        server: Optional[str] = None,
+        protocol: Optional[str] = None,
+        api: Optional[Literal[1, 3]] = None,
+        certificate_validation: Optional[bool] = None,
+        token_refresh_ms: int = 10 * 60 * 1000,
+        store_token_creds_in_memory: Optional[bool] = None,
+        client_protocol_hostname: Optional[str] = None,
+        org_name: Optional[str] = None,
+        idp_name: Optional[str] = None,
+        is_sso_login: Optional[bool] = False,
+        sso_timeout: Optional[int] = SSO_GET_TOKEN_ELAPSE_SECONDS,
+        sso_opt_into_type: Optional[Literal["display", "browser"]] = None 
+):
+        """display HTML link for databricks notebooks
+
+        see graphistry.register docstring for details on input parameters. 
+        """
+
+        PyGraphistry.api_version(api)
+        PyGraphistry.api_token_refresh_ms(token_refresh_ms)
+        PyGraphistry.server(server)
+        PyGraphistry.protocol(protocol)
+        PyGraphistry.client_protocol_hostname(client_protocol_hostname)
+        PyGraphistry.certificate_validation(certificate_validation)
+        PyGraphistry.store_token_creds_in_memory(store_token_creds_in_memory)
+        # Reset token creds
+        PyGraphistry.__reset_token_creds_in_memory()
+
+        # the following code was copied from sso_login() to get the auth_url 
+
+        PyGraphistry._is_authenticated = False
+        arrow_uploader = ArrowUploader(
+            server_base_path=PyGraphistry.protocol()
+            + "://"                     # noqa: W503
+            + PyGraphistry.server(),    # noqa: W503
+            certificate_validation=PyGraphistry.certificate_validation(),
+        ).sso_login(org_name, idp_name)
+
+        try:
+            if arrow_uploader.token:
+                PyGraphistry.api_token(arrow_uploader.token)
+                PyGraphistry._is_authenticated = True
+                arrow_uploader.token = None
+                return PyGraphistry.api_token()
+        except Exception:  # required to log on
+            PyGraphistry.sso_state(arrow_uploader.sso_state)
+            auth_url = arrow_uploader.sso_auth_url
+            if auth_url and not PyGraphistry.api_token():
```

Comment:
in the notebook case (but not needed in dashboard), we risk having an expired token (e.g., been up for an hour)

normally there's an auto-refresh that kicks in for some of the dispatched methods, so this gets auto-handled at plot

BUT when an explicit public call like this one, it makes sense to show the url either way, or to make this an option?